src/pybind/mgr/cephadm/module.py

```diff
@@ -2564,3 +2580,33 @@ def remove_osds_status(self) -> List[OSD]:
         The CLI call to retrieve an osd removal report
         """
         return self.to_remove_osds.all_osds()
+
+    @handle_orch_error
+    def drain_host(self, hostname):
+        # type: (str) -> str
+        """
+        Drain all daemons from a host.
+
+        :param host: host name
+        """
+        self.add_host_label(hostname, '_no_schedule')
+
+        daemons: List[orchestrator.DaemonDescription] = self.cache.get_daemons_by_host(hostname)
+
+        osds_to_remove = []
+        for dd in daemons:
+            self.log.info('DPIVOKA %s %s' % (dd.daemon_type, dd.daemon_id))
+            if dd.daemon_type == 'osd':
+                osds_to_remove.append(dd.daemon_id)
+
+        self.remove_osds(osds_to_remove)
+
+        daemons = [(d.daemon_type, str(d.daemon_id))
+                   for d in daemons]
+        daemons_table = ""
+        daemons.insert(0, ("-" * 20, "-" * 15))
+        daemons.insert(0, ("type", "id"))
+        for d in daemons:
+            daemons_table += "{:<20} {:<15}\n".format(d[0], d[1])
```

Comment:
```suggestion
            daemons_table += "{:<20} {:<15}\n".format(d.daemon_type, d.daemon_id)
```
Is it necessary to reformat daemons into the list of pair values instead of just accessing the fields